### PR TITLE
feat(Tabs): Add

### PR DESCRIPTION
**Pull request by ArthusLiang**
Tabs
Tab
TabList
TabContent
TabHeader
TabPane

**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines: https://github.com/collab-ui/collab-ui-angular/blob/master/CONTRIBUTING.md#commit-message-format
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
- [ ] Bugfix
- [ ] Feature
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?** (check one with "x")
- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: src/lib/tabs/tabs.component.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  template: `
+    <ng-content></ng-content>
+  `,
+  styles: [],
+  providers: [ TabsService ]
+})
+export class TabsComponent implements OnInit {
+
+  /** @prop Optional CSS class name | '' */
+  @Input() class: string = '';
```
Comment (mohammedzamakhan) [inline src/lib/tabs/tabs.component.ts]: I created a Issue https://github.com/collab-ui/collab-ui-angular/issues/19 for why I think `@Input() class` should not be used

File: src/lib/public_api.ts [hunk 2]
```diff
@@ -8,3 +8,10 @@ export * from './list-item-section';
 export * from './spinner';
 export * from './loading';
 export * from './button';
+
+export * from './tab';
+export * from './tab-content';
+export * from './tab-header';
+export * from './tab-list';
+export * from './tab-pane';
+export * from './tabs';
```
Comment (mohammedzamakhan) [inline src/lib/public_api.ts]: Maybe having all these components in `tab`/`tabs` folder makes sense?

File: src/lib/tab/tab.component.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+}
+
+/**
+ * @component tab
+ * @section ng-content
+ * @angular
+ *
+<cui-tabs>
+    <cui-tab-list>
+        <cui-tab class='helloClass'>A</cui-tab>
```
Comment (mohammedzamakhan) [inline src/lib/tab/tab.component.ts]: Also make sure `ngClass` works here

File: src/lib/tabs/tabs.component.ts (hunk 1 above)
Comment (ArthusLiang) [reply src/lib/tabs/tabs.component.ts]: Has checked it in tabs

File: src/lib/public_api.ts (hunk 2 above)
Comment (ArthusLiang) [reply src/lib/public_api.ts]: Check the updates

**Comment by ArthusLiang** [comment]
> I ran the playground, and found that tabs are not keyboard accessible, where you can use up/down/left/right arrow keys to navigate between the tabs. Use `FocusKeyManager` from `@angular/cdk` to probably achieve it.

Add keypress feature with native javascript. You can even switch between tabs with the same first letter.

**Comment by mohammedzamakhan** [comment]
@ArthusLiang I just testing it out with voiceover and saw the following issues (which does not exists in react collab):

When I use either arrows for the first letter, I see that the intended tab is selected (just using the focus color), but not actually selected and thus I see that

1) The tab is not read out by voiceover.
2) It selects the tab, which it should not (it should just focus, not select)
3) When I press enter, it does not select the tab should have focus style, but it selects the first tab

**Comment by mohammedzamakhan** [comment]
Still not able to add a class using ngClass or either by passing class attribute

**Pull request merged**